hyperion-geiger-counter/src/main/java/com/willowtreeapps/hyperion/geigercounter/GeigerCounterModule.java

```diff
@@ -0,0 +1,70 @@
+package com.willowtreeapps.hyperion.geigercounter;
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.willowtreeapps.hyperion.plugin.v1.PluginModule;
+
+class GeigerCounterModule extends PluginModule implements View.OnClickListener {
+
+    private View view;
+    private DroppedFrameObserver observer;
+
+    // Helpers
+
+    private static boolean arePrerequisitesAvailable() {
+        return DroppedFrameObserver.TARGET_API <= Build.VERSION.SDK_INT;
+    }
+
+    // PluginModule
+
+    @Override
+    protected void onCreate() {
+        if (!arePrerequisitesAvailable()) {
+            return;
+        }
+
+        observer = new DroppedFrameObserver(getExtension().getActivity());
+    }
+
+    @Nullable
+    @Override
+    public View createPluginView(@NonNull LayoutInflater layoutInflater, @NonNull ViewGroup parent) {
```

Comment:
PluginModules are created per Activity, but the framerate tracker is active per Application (Choreographer.getInstance()). What this means is that if you activate the plugin from Activity A and then navigate to Activity B, the menu embedded in Activity B will show the plugin view as unselected even though the tracker is on. Probably what we wanna do is have the Observer exist at a higher scope and be passed into the Module. Plugins are singleton, so probably that should work fine. Then when you create the view, immediately update the state to reflect the state of the observer, since it may be active from a previous activity.

Reply:
OK, check my understanding: The Plugin can own one Observer, and each instantiated Module can use the shared Observer and watch it in order to keep their appearance updated.